internal/editor/editor.go

```diff
@@ -0,0 +1,126 @@
+	}
+
+	return editorAndArguments[0], args
+}
+
+// edit create a temporary file with given content, start a text editor then return edited content
+// temporary file will be deleted on complete
+// temporary file is not deleted if edit fails
+func edit(content []byte) ([]byte, error) {
+	if SkipEditor {
+		return content, nil
+	}
+
+	tmpFileName, err := createTemporaryFile(content, marshalMode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temporary file: %w", err)
+	}
+	defer os.Remove(tmpFileName)
+
+	editorPath, args := editorPathAndArgs(tmpFileName)
+	cmd := exec.Command(editorPath, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+
+	err = cmd.Run()
+	if err != nil {
+		return nil, fmt.Errorf("failed to edit temporary file %q: %w", tmpFileName, err)
+	}
+
+	editedContent, err := os.ReadFile(tmpFileName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read temporary file %q: %w", tmpFileName, err)
+	}
+
+	return editedContent, nil
+}
+
+// updateResourceEditor takes a complete resource and a partial updateRequest
+// will return a copy of updateRequest that has been edited
+func updateResourceEditor(resource interface{}, updateRequest interface{}, cfg *Config) (interface{}, error) {
+	// Create a copy of updateRequest completed with resource content
+	completeUpdateRequest := copyAndCompleteUpdateRequest(updateRequest, resource)
+
+	// TODO: fields present in updateRequest should be removed from marshal
+	// ex: namespace_id, region, zone
+	// Currently not an issue as fields that should be removed are mostly path parameter /{zone}/namespace/{namespace_id}
+	// Path parameter have "-" as json tag and are not marshaled
+
+	updateRequestMarshaled, err := marshal(completeUpdateRequest, cfg.MarshalMode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal update request: %w", err)
+	}
+
+	// Start text editor to edit marshaled request
+	updateRequestMarshaled, err = edit(updateRequestMarshaled)
+	if err != nil {
+		return nil, fmt.Errorf("failed to edit marshalled data: %w", err)
+	}
+
+	// If editedResource is present, override edited resource
+	// This is useful for testing purpose
+	if cfg.editedResource != "" {
+		updateRequestMarshaled = []byte(cfg.editedResource)
+	}
+
+	// Create a new updateRequest as destination for edited yaml/json
+	// Must be a new one to avoid merge of maps content
+	updateRequestEdited := newRequest(updateRequest)
+
+	// TODO: if !putRequest
+	// fill updateRequestEdited with only edited fields and fields present in updateRequest
+	// fields should be compared with completeUpdateRequest to find edited ones
+
+	// Add back required non-marshaled fields (zone, ID)
+	copyRequestPathParameters(updateRequestEdited, updateRequest)
+
+	err = unmarshal(updateRequestMarshaled, updateRequestEdited, cfg.MarshalMode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal edited data: %w", err)
+	}
+
+	return updateRequestEdited, nil
```

Comment:
When I'm not editing I'm stumbling upon:

```
$ go run ./cmd/scw instance security-group edit bc5dd974-1206-4777-aa93-e71c6c2a3eb2
Rules.0.ID            58909be7-d17c-4ac8-9eb3-23d5fc58abc5
Rules.0.Protocol      TCP
Rules.0.Direction     outbound
Rules.0.Action        drop
Rules.0.IPRange       0.0.0.0/0
Rules.0.DestPortFrom  25
Rules.0.Position      1
Rules.0.Editable      false
Rules.0.Zone          fr-par-1
Rules.1.ID            25680235-108b-4bbc-8e25-114303d950bd
Rules.1.Protocol      TCP
...
```

Instead of the nice visualisation that I get if I do a get commands:

```
$ go run ./cmd/scw instance security-group get bc5dd974-1206-4777-aa93-e71c6c2a3eb2 
Security Group:
ID                     bc5dd974-1206-4777-aa93-e71c6c2a3eb2
Name                   cli-sg-pensive-knuth
State                  available
Description            -
EnableDefaultSecurity  true
OrganizationID         951df375-e094-4d26-97c1-ba548eeb9c42
ProjectID              951df375-e094-4d26-97c1-ba548eeb9c42
OrganizationDefault    false
ProjectDefault         false
CreationDate           8 minutes ago
ModificationDate       18 seconds ago
Stateful               true

Inbound Rules (default policy accept):
ID                                    PROTOCOL  ACTION  IP RANGE   DEST
72d3f498-1579-4485-a79a-ceaafebb79c4  UDP       accept  0.0.0.0/0  21
52b111fe-2b36-4ec9-96bc-0aa70cb2cea2  UDP       accept  0.0.0.0/0  22
8bc9936c-f29a-40e8-8781-1c7aa058a84c  UDP       accept  0.0.0.0/0  23

Outbound Rules (default policy accept):
ID                                    PROTOCOL  ACTION  IP RANGE   DEST
58909be7-d17c-4ac8-9eb3-23d5fc58abc5  TCP       drop    0.0.0.0/0  25
25680235-108b-4bbc-8e25-114303d950bd  TCP       drop    0.0.0.0/0  465
4a31b633-118e-4900-bd52-facf1085fc8d  TCP       drop    0.0.0.0/0  587
e7dd28e8-3747-4c7c-9a4f-35ae3f0ae2cd  TCP       drop    ::/0       25
f37d9e7c-8ed7-4e0f-baff-7f5e7ede0baf  TCP       drop    ::/0       465
68054851-54e3-46c9-9cd7-83219751248b  TCP       drop    ::/0       587
e46c7bc0-7da1-4ac8-b559-ee7207d1cf96  UDP       accept  0.0.0.0/0  21

Servers:
ID   NAME
```

Reply:
Fixed with #2939